src/lib/stores/decrypt.ts

```diff
@@ -0,0 +1,45 @@
+import { browser } from '$app/env';
+import { writable } from 'svelte/store';
+
+export const decryptPort = persistStore('port', null);
+
+// An enum or string union may be more appropriate here so we dont' get a
+// if thing and the falsiness evaluates to the null case, but start with this
+export const lastDecryptStatus = writable<null | boolean>(null);
+
+export const portNumber = writable<null | number>(null);
+
+export function lastDecryptFailure() {
+  lastDecryptStatus.set(false);
+}
+
+export function lastDecryptSuccess() {
+  lastDecryptStatus.set(true);
+}
+
+export function persistStore(name, initialValue) {
```

Comment:
Do you like this name?

Reply:
No, but I couldn't figure a better one. Open to suggestions

Reply:
`persistentStore` `persistStoreToLocalStorage` `localStoragStore` are all I've got